dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,580 @@
+        self.best_params_ = cv_results["params"][best_index]
+        self.scorer_ = scorer
+
+        self.model_history_ = estimator_history
+        self.history_ = history
+        self.cv_results_ = cv_results
+
+        self.multimetric_ = SHAs[best_bracket].multimetric_
+        self._SuccessiveHalvings_ = SHAs
+        raise gen.Return(self)
+
+    @property
+    def metadata(self):
+        bracket_info = _hyperband_paper_alg(self.max_iter, eta=self.aggressiveness)
+        num_models = sum(b["estimators"] for b in bracket_info)
+        for bracket in bracket_info:
+            bracket["decisions"].update({1})
+            bracket["decisions"] = sorted(list(bracket["decisions"]))
+        num_partial_fit = sum(b["partial_fit_calls"] for b in bracket_info)
+        bracket_info = list(reversed(sorted(bracket_info, key=lambda x: x["bracket"])))
+
+        brackets = _get_hyperband_params(self.max_iter, eta=self.aggressiveness)
+        SHAs = self._get_SHAs(brackets)
+        for bracket in bracket_info:
+            b = bracket["bracket"]
+            bracket["SuccessiveHalvingSearchCV params"] = _get_SHA_params(SHAs[b])
+
+        bracket_info = sorted(bracket_info, key=lambda x: x["bracket"])
+        info = {
+            "partial_fit_calls": num_partial_fit,
+            "estimators": num_models,
+            "brackets": bracket_info,
+        }
+        return info
+
+
+def _get_meta(hists, brackets, SHAs, key):
+    meta_ = []
+    history_ = {}
+    for bracket in brackets:
+        hist = hists[bracket]
+
+        info_hist = {key(bracket, h["model_id"]): [] for h in hist}
+        for h in hist:
+            info_hist[key(bracket, h["model_id"])] += [h]
+        hist = info_hist
+        history_.update(hist)
+
+        calls = {k: max(hi["partial_fit_calls"] for hi in h) for k, h in hist.items()}
+        decisions = {hi["partial_fit_calls"] for h in hist.values() for hi in h}
+        meta_.append({
+            "decisions": sorted(list(decisions)),
+            "estimators": len(hist),
+            "bracket": bracket,
+            "partial_fit_calls": sum(calls.values()),
+            "SuccessiveHalvingSearchCV params": _get_SHA_params(SHAs[bracket]),
+        })
+    meta_ = sorted(meta_, key=lambda x: x["bracket"])
+    return meta_, history_
+
+
+def _get_SHA_params(SHA):
+    """
+    Parameters
+    ----------
+    SHA : SuccessiveHalvingSearchCV
+
+    Returns
+    -------
+    params : dict
+        Dictionary to re-create a SuccessiveHalvingSearchCV without the
+        estimator or parameters
+
+    Example
+    -------
+    >>> from sklearn.linear_model import SGDClassifier
+    >>> model = SGDClassifier()
+    >>> params = {"alpha": np.logspace(-1, 1)}
+    >>> SHA = SuccessiveHalvingSearchCV(model, params, tol=0.1,
+    ...                                 patience=True, random_state=42)
+    >>> _get_SHA_params(SHA)
+    {'aggressiveness': 3,
+     'max_iter': 100,
+     'n_initial_iter': 9,
```

Comment:
Can this `n_initial_iter` in the SHA be removed? Instead, can `n_iter` be specified?